cheatsheets/File_Upload_Cheat_Sheet.md

```diff
@@ -0,0 +1,163 @@
+# File Upload Cheat Sheet
+
+## Introduction
+
+File upload is becoming more and more an essential part of any application, where the user is able to upload their photo, their CV, or a video showcasing a project they are working on. It is evident that applications should be able to handle file uploads based on the business requirement. In return, the application should be able to fend off bogus, and malicious files in a way to keep the application and the users safe.
+
+In short, the following principles should be followed to reach a secure file upload implementation:
+
+- **Whitelist allowed extensions. Only allow safe and critical extensions for business functionality**
+  - **Ensure that [input validation](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Input_Validation_Cheat_Sheet.md#validating-free-form-unicode-text) is applied before validating the extensions.**
+- **Allowed users only can upload files**
```

Comment:
`Only allow authorised users to upload files`

Reply:
Also maybe move this down, so that all the file type/content points are together?